<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.47">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>specifications – Soil Health Data Cube for Europe</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<meta property="og:title" content="Soil Health Data Cube for Europe">
<meta property="og:description" content="Soil Health Data Cube for pan-EU (SHDC4EU) is compilation of pan-EU layers organized into a data cube and served through an API as analysis or decision ready data.">
<meta property="og:site_name" content="Soil Health Data Cube for Europe">
<meta name="twitter:title" content="Soil Health Data Cube for Europe">
<meta name="twitter:image" content="images/cover.png">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="././images/logo_ai4soilhealth_square.bw.short.png" alt="" class="navbar-logo">
    </a>
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Soil Health Data Cube for Europe</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/ai4soilhealth"> 
<span class="menu-text">Project repositories</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://ai4soilhealth.eu"> 
<span class="menu-text">AI4SoilHealth project</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools tools-wide">
    <a href="https://fosstodon.org/@opengeohub" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-mastodon"></i></a>
    <a href="https://opengeohub.medium.com/" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-medium"></i></a>
    <a href="https://twitter.com/AI4SoilHealth" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-twitter"></i></a>
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" role="link" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/AI4SoilHealth/SHDC4EU_manual">
            Source Code
            </a>
          </li>
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/AI4SoilHealth/SHDC4EU_manual/issues">
            Report a Bug
            </a>
          </li>
      </ul>
    </div>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./index.html">Soil Health Data Cube components</a></li><li class="breadcrumb-item"><a href="./1-specifications.html">Soil Health Data Cube specifications</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Soil Health Data Cube components</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./preface.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1-specifications.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Soil Health Data Cube specifications</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2-points.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Point data import and quality control</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Data use tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#soil-health-data-cube-specifications" id="toc-soil-health-data-cube-specifications" class="nav-link active" data-scroll-target="#soil-health-data-cube-specifications">Soil Health Data Cube specifications</a>
  <ul class="collapse">
  <li><a href="#the-7-step-framework-for-soil-health-assessment" id="toc-the-7-step-framework-for-soil-health-assessment" class="nav-link" data-scroll-target="#the-7-step-framework-for-soil-health-assessment">The 7-step framework for soil health assessment</a></li>
  <li><a href="#predictive-soil-mapping-based-on-machine-learning-and-hpc" id="toc-predictive-soil-mapping-based-on-machine-learning-and-hpc" class="nav-link" data-scroll-target="#predictive-soil-mapping-based-on-machine-learning-and-hpc">Predictive soil mapping based on Machine Learning and HPC</a></li>
  <li><a href="#training-points" id="toc-training-points" class="nav-link" data-scroll-target="#training-points">Training points</a></li>
  <li><a href="#covariate-base-layers" id="toc-covariate-base-layers" class="nav-link" data-scroll-target="#covariate-base-layers">Covariate (base) layers</a></li>
  <li><a href="#predictive-mapping-and-derivation-methods" id="toc-predictive-mapping-and-derivation-methods" class="nav-link" data-scroll-target="#predictive-mapping-and-derivation-methods">Predictive mapping and derivation methods</a></li>
  <li><a href="#targeted-soil-and-soil-health-variables" id="toc-targeted-soil-and-soil-health-variables" class="nav-link" data-scroll-target="#targeted-soil-and-soil-health-variables">Targeted soil and soil-health variables</a></li>
  <li><a href="#variable-registry" id="toc-variable-registry" class="nav-link" data-scroll-target="#variable-registry">Variable registry</a></li>
  <li><a href="#spatio-temporal-reference" id="toc-spatio-temporal-reference" class="nav-link" data-scroll-target="#spatio-temporal-reference">Spatio-temporal reference</a></li>
  <li><a href="#uncertainty-of-predictions" id="toc-uncertainty-of-predictions" class="nav-link" data-scroll-target="#uncertainty-of-predictions">Uncertainty of predictions</a></li>
  <li><a href="#back-end-front-end-components" id="toc-back-end-front-end-components" class="nav-link" data-scroll-target="#back-end-front-end-components">Back-end / front-end components</a></li>
  <li><a href="#data-exchange-formats" id="toc-data-exchange-formats" class="nav-link" data-scroll-target="#data-exchange-formats">Data exchange formats</a></li>
  <li><a href="#standard-spatialtemporal-resolutions-and-support-sizes" id="toc-standard-spatialtemporal-resolutions-and-support-sizes" class="nav-link" data-scroll-target="#standard-spatialtemporal-resolutions-and-support-sizes">Standard spatial/temporal resolutions and support sizes</a></li>
  <li><a href="#file-naming-convention" id="toc-file-naming-convention" class="nav-link" data-scroll-target="#file-naming-convention">File Naming convention</a></li>
  <li><a href="#registering-project-outputs" id="toc-registering-project-outputs" class="nav-link" data-scroll-target="#registering-project-outputs">Registering project outputs</a></li>
  <li><a href="#disclaimer-for-data-software-products" id="toc-disclaimer-for-data-software-products" class="nav-link" data-scroll-target="#disclaimer-for-data-software-products">Disclaimer for data / software products</a></li>
  <li><a href="#land-mask" id="toc-land-mask" class="nav-link" data-scroll-target="#land-mask">Land mask</a></li>
  <li><a href="#layer-submission-process-and-quality-control" id="toc-layer-submission-process-and-quality-control" class="nav-link" data-scroll-target="#layer-submission-process-and-quality-control">Layer submission process and quality control</a></li>
  <li><a href="#reproducible-research" id="toc-reproducible-research" class="nav-link" data-scroll-target="#reproducible-research">Reproducible research</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/AI4SoilHealth/SHDC4EU_manual/edit/main/1-specifications.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/AI4SoilHealth/SHDC4EU_manual/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">




<section id="soil-health-data-cube-specifications" class="level1">
<h1>Soil Health Data Cube specifications</h1>
<section id="the-7-step-framework-for-soil-health-assessment" class="level2">
<h2 class="anchored" data-anchor-id="the-7-step-framework-for-soil-health-assessment">The 7-step framework for soil health assessment</h2>
<p>In a nutshell, Soil Health Data Cube is a building block and infrastructure for monitoring soil health across EU. The general idea of the SHDC4EU is that it serves some standard processes that help inform land owners / land managers about the soil health state of their land using open data. These processes are meant to help future users of the AI4SoilHealth app to quickly find out about the site, explore land potentials based on the state-of-the-art models and start taking concrete actions towards restoring land / improving soil health. This is closely connected with the main expected impacts of the AI4SoilHealth project (to make soil health data easier to access and provide tools and standards for soil health monitorng). In simple terms, the 7–step soil health assessment framework <span class="citation" data-cites="hengl2024osg">(<a href="#ref-hengl2024osg" role="doc-biblioref">Hengl 2024</a>)</span> includes:</p>
<ol type="1">
<li><strong>Assess the history of a site</strong> (last 30+ years) provide answers to questions e.g.:
<ul>
<li>How much SOC has been gained / lost over the last 30 years?</li>
<li>Which chemical and physical soil properties changed the most?</li>
<li>Is there any land/soil degradation and how significant is it?</li>
<li>How has agronomic management impacted soil health?</li>
<li>How much did the land cover / vegetation cover change? What is the main <a href="https://docs.google.com/spreadsheets/d/1SX51OilNt-cUYpAa7t0LAvZRTzq3Sd4WJLnX6mWKfQk/edit#gid=1828976853">change class</a>?</li>
<li>How much did the landscape heterogeneity change over time?</li>
</ul></li>
<li><strong>Determine current state of soil</strong> (actual year):
<ul>
<li>What is the current state of physical and chemical soil properties?</li>
<li>What is the current (WRB) soil type / main soil forming processes and soil states?</li>
<li>What are the current derived soil properties?</li>
<li>What are the current macronutrient stocks in the soil?</li>
</ul></li>
<li><strong>Determine soil potential</strong> (next 10, 20, 30 yrs):
<ul>
<li>What is the SOC sequestration potential of this soil?</li>
<li>What is the agricultural potential of this soil?</li>
<li>What are the other potential ecosystem functions provided some land use change or mitigations?</li>
</ul></li>
<li><strong>Collect extra soil samples, send to lab, and add to training points</strong>:
<ul>
<li>Generate a sampling design that helps increase mapping accuracy. Optimize the number of samples while taking into account the costs of sampling.</li>
<li>Collect new samples and send them to the lab.</li>
</ul></li>
<li><strong>Import new local data then re-analyze and re-asses points 1, 2 and 3</strong>.
<ul>
<li>Import new local samples / results and add to the training points. Re-assess local prediction accuracy.</li>
<li>Repredict layers in SHDC4EU for steps #1, #2 and #3.</li>
</ul></li>
<li><strong>Suggest concrete KPI’s</strong> (per farm / per land unit):
<ul>
<li>Provided that soil potential assessment reveals a significant gap between potential and actual.</li>
<li>Either directly provide concrete measures to help increase soil health and/or provide links to documents / organizations that can directly help increase soil health.</li>
</ul></li>
<li><strong>Track progress per farm and re-assess if required</strong>:
<ul>
<li>Every 1–2 years update the indicators (pan-EU).</li>
<li>Compare with the planned transition in #6. If necessary recommend re-assessment.</li>
</ul></li>
</ol>
<p>This framework can be either apply to assess health of a specific site (either in the field using a mobile phone app, or in the office using a desktop app), or complete farms / administrative units. We, thus, aim to complete the data cube with primary and derived soil variables that can directly serve this framework, and especially so that can be worked alongside other Work Packages in the project in their testing of methodology.</p>
</section>
<section id="predictive-soil-mapping-based-on-machine-learning-and-hpc" class="level2">
<h2 class="anchored" data-anchor-id="predictive-soil-mapping-based-on-machine-learning-and-hpc">Predictive soil mapping based on Machine Learning and HPC</h2>
<p>WP5 — with the tasks T5.2 Development of AI4SoilHealth computing engine and T5.3 Development of AI4SoilHealth Indicators data cube) — aims at implementing fully automated predictive soil mapping at high spatial resolution using large datasets (e.g.&nbsp;whole of Europe at 30-m spatial resolution). Prepared harmonized point datasets will are overlaid against time-series of EO images and/or static terrain-based or similar indices. These are then be used to predict values of target variables in 2D, 3D and 2D+T, 3D+T. The general workflow for predictive soil mapping based on <strong>Spatiotemporal Machine Learning</strong> and <strong>High Performance Computing</strong> (HPC) and assumes that all point data is analysis-ready, representative and correlates with the EO / covariate layers i.e.&nbsp;can be used to produce usable spatial predictions. Automated predictive soil mapping is currently implemented via the Python library <a href="https://github.com/scikit-map/scikit-map">scikit-map</a> using OpenGeoHub’s High Performance Computing infrastructure.</p>
<p>The key interest of WP5 is to produce pan-EU predictions at the highest possible spatial resolution and that can directly serve soil health assessment at continental scale. Some variables are, however, not available for the whole of pan-EU, and some point datasets will be produced experimentally and used for testing purposes only. Hence, within the WP5, all soil variables are modeled / mapped under one the following 3 tiers:</p>
<ul>
<li><strong>Tier 1: Pan-EU, production ready variables</strong>: usually based on <a href="https://esdac.jrc.ec.europa.eu/projects/lucas">LUCAS + national soil datasets</a>;</li>
<li><strong>Tier 2 (internal, under construction): National / in-situ data</strong>: Tested locally, then upscaled to whole of EU; usually based on collaboration of WP3,4,5,6;</li>
<li><strong>Tier 3 (internal, under construction): Pan-EU, new experimental variables</strong> (currently not in the LUCAS soil and most likely can not be modeled/mapped across EU, but can only be used to assess soil health <em>at site</em>); based on the collaboration of WP3,4,5;</li>
</ul>
<p>We expect that project partners + pilots will feed data for Tier 2 and Tier 3. However, not everything can be mapped across the EU, hence some variables will eventually stay available only locally and a small selection of variables will be available <em>in-situ</em> only i.e.&nbsp;these will not be mapped at all.</p>
</section>
<section id="training-points" class="level2">
<h2 class="anchored" data-anchor-id="training-points">Training points</h2>
<p>Quality of the outputs would most likely be dictated by the following four key aspects of modeling:</p>
<ol type="1">
<li><strong>The general modeling design</strong> i.e.&nbsp;how appropriate the statistical / machine learning methods are. For example: are all casualties considered, are all distributions represented, are all statistical assumptions met?</li>
<li><strong>Predictive performance of the algorithms used</strong> i.e.&nbsp;is the best / fastest algorithm used for modeling? Is the algorithm robust, noise-proof, artifacts-proof etc?</li>
<li><strong>Quality and diversity of the covariate layers used</strong>.</li>
<li><strong>Quality and spatial and feature space representation of training points</strong>.</li>
</ol>
<p>Assuming that #1 and #2 are optimized, then the only remaining factors that control success of the modeling / mapping processes are aspects #3 and #4 i.e.&nbsp;the quality of covariate layers and quality of training points. In the past (prior to 2010), it was relatively difficult to produce pan-EU maps of soil health indicators as there was only limited training point data. Thanks to the European Commission’s <a href="https://esdac.jrc.ec.europa.eu/projects/lucas">LUCAS soil project</a>, we now have 3–4 repetitions of ground measurements of soil properties of highest quality (about 22,000 sites are revisited per period) (see figure below). This is a unique opportunity to test building high resolution predictions, including dynamic soil property predictions. Ideally, such testing should be a joint effort of the AI4SoilHealth consortium, in an open collaboration with JRC and members of other sister projects funded by the same Soil mission.</p>
<p>Nevertheless, LUCAS soil is also not an ideal dataset for predictive mapping. Hence producing SHDC4EU faces two serious challanges: (1) LUCAS soil is a top-soil dataset, although there is now intention to sample also subsoils, (2) it covers only a number of physical and chemical soil variables (10–15), and soil types or similar are typically not recorded, making this largely a partial pedological survey.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/fig_lucas_points_scheme.jpg" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption>LUCAS soil samples <span class="citation" data-cites="orgiazzi2018lucas">(<a href="#ref-orgiazzi2018lucas" role="doc-biblioref">Orgiazzi et al. 2018</a>)</span> and connected existing and upcoming EO missions (bars indicate approximated temporal coverage). Note that the amount of EO data and missions is increasing exponentially.</figcaption>
</figure>
</div>
</div>
</div>
<p>To improve usability of predictions produced using LUCAS we have decided, in this project, to combine both LUCAS and the highest quality legacy national soil datasets. These synchronized and analysis ready fusions of soil laboratory points and observations will be generated by T4.6 “Integration and harmonization of in-situ and ancillary observations”. We anticipate that this will be a long process hence, in order to prevent serious delays, we will start producing SHDC4EU predictions, then in each iteration try to improve predictions as new countries join the campaign of contributing their data (under a standard Data Sharing Agreement) for the purpose of data mining i.e.&nbsp;producing open soil information for everyone.</p>
</section>
<section id="covariate-base-layers" class="level2">
<h2 class="anchored" data-anchor-id="covariate-base-layers">Covariate (base) layers</h2>
<p>The second important aspect that will determine the quality of the SHDC4EU outputs is the quality of covariate layers. Quality of the covariate layers is basically determined by three things: (1) spatial and temporal resolution, (2) spectral / thematic content, (3) general quality of data in terms of completeness, consistency, correctness and quality of metadata. For producing SHDC4EU we plan to use an extensive compilation of covariate (base) layers to produce predictions for SHDC4EU. Already available layers for continental Europe include (<a href="https://EcoDataCube.eu">https://EcoDataCube.eu</a>):</p>
<ul>
<li>Land mask (pan-EU: <a href="https://zenodo.org/doi/10.5281/zenodo.8171860">https://zenodo.org/doi/10.5281/zenodo.8171860</a>) + World Settlement Footprint (WSF) i.e.&nbsp;world buildings 2000-2015, 2019 (<a href="https://geoservice.dlr.de/web/maps/eoc:wsfevolution">https://geoservice.dlr.de/web/maps/eoc:wsfevolution</a> and <a href="https://geoservice.dlr.de/web/maps/eoc:wsf2019">https://geoservice.dlr.de/web/maps/eoc:wsf2019</a>);</li>
<li>Digital Terrain Model variables at 6–scales 30-m, 60, 120, 240, 480 and 960-m:
<ul>
<li>Elevation, slope (%), min- max-curvature,</li>
<li>Hillshade, northness, easterness,</li>
<li>Positive, negative openness,</li>
<li>Topidx (TWI), geomorphon classes (10), catchment area, LS factor,</li>
</ul></li>
<li>Lithological (surface geology) map of pan-EU at 250-m (<a href="https://zenodo.org/doi/10.5281/zenodo.4787631">https://zenodo.org/doi/10.5281/zenodo.4787631</a>);</li>
<li>Bimonthly / quarterly GLAD Landsat composites (all bands + biophysical indices, 2000–2022) (explained in: <a href="https://doi.org/10.21203/rs.3.rs-4251113/v1">https://doi.org/10.21203/rs.3.rs-4251113/v1</a>):
<ul>
<li>Green, Red, NIR, SWIR1, SWIR2,</li>
<li>NDVI, FAPAR,</li>
<li>NDTI, NDWI, BSF,</li>
<li>Cumulative NDVI, NDTI and BSF;</li>
</ul></li>
<li>Climatic variables at 1-km (long-term or monthly series):
<ul>
<li>CHELSA Climate Bioclimatic variables (<a href="https://chelsa-climate.org/bioclim/">https://chelsa-climate.org/bioclim/</a>);</li>
<li>CHELSA monthly precipitation time-series 2000–2022 (<a href="https://chelsa-climate.org/timeseries/">https://chelsa-climate.org/timeseries/</a>);</li>
<li>MODIS LST daytime and nighttime monthly time-series 2000–2022 (<a href="https://zenodo.org/doi/10.5281/zenodo.1420114">https://zenodo.org/doi/10.5281/zenodo.1420114</a>);</li>
<li>MODIS monthly water vapor time-series 2000–2022 (<a href="https://zenodo.org/doi/10.5281/zenodo.8193738">https://zenodo.org/doi/10.5281/zenodo.8193738</a>);</li>
<li>Cumulative annual precipitation (2000–2022);</li>
</ul></li>
<li>Monthly / annual snow images 2000-2022 e.g.&nbsp;DLR Snow pack at 500-m (<a href="https://geoservice.dlr.de/web/maps/eoc:gsp:yearly">https://geoservice.dlr.de/web/maps/eoc:gsp:yearly</a>):
<ul>
<li>Long-term monthly snow probability time-series at 500-m (<a href="https://doi.org/10.5281/zenodo.5774953">https://doi.org/10.5281/zenodo.5774953</a>);</li>
<li>Cumulative annual snow probability 2000–2022;</li>
</ul></li>
<li>Surface water dynamics: occurrence probability of water long-term 1999–2021 (<a href="https://glad.umd.edu/dataset/global-surface-water-dynamics">https://glad.umd.edu/dataset/global-surface-water-dynamics</a>);</li>
<li>Optional: Global Flood Database v1 (2000-2018) at 250-m annual flood event (<a href="http://global-flood-database.cloudtostreet.info/">http://global-flood-database.cloudtostreet.info/</a>);</li>
<li>Light at night time-series at 500-m resolution 2000–2022 (<a href="https://zenodo.org/doi/10.5281/zenodo.7750174">https://zenodo.org/doi/10.5281/zenodo.7750174</a>);</li>
<li><a href="https://zenodo.org/doi/10.5281/zenodo.12527545">Cropland extent</a> at 30-m for 2000 to 2022 based on (<a href="https://glad.umd.edu/dataset/croplands">https://glad.umd.edu/dataset/croplands</a>);</li>
<li><a href="https://zenodo.org/doi/10.5281/zenodo.11961219">Bare soil percent</a> and photosythetical vegetation percent annual for 2000 to 2022 based on the MODIS MCD43A4 product (500-m spatial resolution);</li>
</ul>
<p>In addition to the existing layers above, we will also generate a number of novel layers tailored specifically for the purpose of representing land use practices and potential soil degradation factors. This include:</p>
<ul>
<li>BSI, <strong>Bare Soil Index</strong> <span class="citation" data-cites="mzid2021analysis">(<a href="#ref-mzid2021analysis" role="doc-biblioref">Mzid et al. 2021</a>)</span> and annual BEF, <strong>Bare Earth Fraction</strong> (e.g.&nbsp;proportion of pixels with NDVI &lt;0.35 based on 16–day and/or bimonthly data; there are other possible thresholds combination that can be used to optimize the results),</li>
<li>NDTI, <strong>Normalized Differential Tillage Index</strong> <span class="citation" data-cites="ettehadi2019separating">(<a href="#ref-ettehadi2019separating" role="doc-biblioref">Ettehadi Osgouei et al. 2019</a>)</span>,</li>
<li>NOS, <strong>Number of Seasons</strong> / NOCC,** Number of Cropping Cycles**,</li>
<li>LOS, <strong>Length of Seasons</strong> / CDR, <strong>Crop Duration Ratio</strong> <span class="citation" data-cites="estel2016mapping">(<a href="#ref-estel2016mapping" role="doc-biblioref">Estel et al. 2016</a>)</span>,</li>
<li>Crop-type based on <a href="https://www.eurocrops.tum.de/">EuroCrops dataset</a>,</li>
</ul>
<p>Note that from all covariate layers listed above, 16-day / bimonthly / quarterly GLAD Landsat composites (all bands + all indices, 2000–2022) are the largest part of the data to be used taking almost 20TB of storage (in compressed format).</p>
</section>
<section id="predictive-mapping-and-derivation-methods" class="level2">
<h2 class="anchored" data-anchor-id="predictive-mapping-and-derivation-methods">Predictive mapping and derivation methods</h2>
<p>Based on the availability of the training points and nature of the target variable in the SHDC4EU, we will either predict or derive soil variables from primary variables. The output soil health indicators can, thus, be considered of type either:</p>
<ol type="1">
<li><strong>Predicted dynamic pan-EU variables</strong> (2000–2022+) available at standard depth intervals (0–20, 20–50, 50–100, 100–200 cm).</li>
<li><strong>Predicted static pan-EU variables</strong>.</li>
<li><strong>Predicted and/or simulated pan-EU variables</strong> i.e.&nbsp;results of predictions of future conditions, scenario testing, simulations, process-based modeling.</li>
<li><strong>Derived pan-EU variables and indices</strong>.</li>
<li><strong>In-situ only variables</strong>.</li>
</ol>
<p>Most of the soil chemical and biological variables are mapped as dynamic variables with annual or 5–year estimates (e.g.&nbsp;5 maps for period 2000–2025) and at multiple depths <span class="citation" data-cites="witjes2023ecodatacube">(<a href="#ref-witjes2023ecodatacube" role="doc-biblioref">Witjes et al. 2023</a>)</span>. This means that the amount of produced data can be significant if predictions are also provided per depth. For example, to map soil organic carbon content (weight %), we can predict 23 annual maps at 4–5 standard depths resulting in over 180 images (assuming that we also produce prediction errors per pixel). AI4SoilHealth tasks T5.5 “Development of AI4SoilHealth present and future soil degradation products”, T5.6 “Development of AI4SoilHealth forecasting services”, and T5.7 “Development of AI4SoilHealth soil functions evidence / model chains” will also generate large amounts of predicted and/or simulated pan-EU variables i.e.&nbsp;results of predictions of future conditions, scenario testing, simulations, process-based modeling.</p>
<p>The following predictive soil modeling methods are considered for use so far for generating SHDC4EU:</p>
<ol type="1">
<li>2D and 3D predictive soil mapping (static predictions, time is ignored).</li>
<li>2D+T predictive soil mapping spacetime.</li>
<li>3D+T predictive soil mapping spacetime.</li>
<li>Predictive soil mapping focused on long-term indices.</li>
<li>Derivation from primary data.</li>
<li>Derivation using Pedo-transfer functions.</li>
<li>Derivation / simulations using Mechanistic model (iterative).</li>
<li>Derivation of future projected predictions (scenario testing).</li>
</ol>
<p>For each derivation method we use sets of algorithms, usually implemented in python or R programming languages. These are documented in detail and eventually allow for complete reproducibility of results; most importantly we do a series of benchmarking (listed in further sections) to compare predictive performance and select the algorithm that is most accurate + most robust at the same time. In the tasks T5.5 Development of AI4SoilHealth present and future soil degradation products, T5.6 Development of AI4SoilHealth forecasting services and T5.7 Development of AI4SoilHealth soil functions evidence / model chains, also <strong>process-based modeling</strong> can be used, especially to generate potential soil ecosystem services etc. Such modeling is at the order-of-magnitude more computationally demanding than predictive mapping hence we expect that these outputs will be of limited spatial detail (1-km) and or available only for experimental testing.</p>
</section>
<section id="targeted-soil-and-soil-health-variables" class="level2">
<h2 class="anchored" data-anchor-id="targeted-soil-and-soil-health-variables">Targeted soil and soil-health variables</h2>
<p>The general objective of the SHDC4EU is to serve best possible, most detailed, complete and consistent predictions of the the number of targeted soil health indicators in the EU Mission’s <a href="https://research-and-innovation.ec.europa.eu/funding/funding-opportunities/funding-programmes-and-open-calls/horizon-europe/eu-missions-horizon-europe/soil-deal-europe_en">“Implementation Plan: A Soil Deal for Europe”</a>:</p>
<ol type="1">
<li>presence of pollutants, excess nutrients and salts,</li>
<li>soil organic carbon stock,</li>
<li>soil structure including soil bulk density and absence of soil sealing and erosion,</li>
<li>soil biodiversity,</li>
<li>soil nutrients and acidity (pH),</li>
<li>vegetation cover,</li>
<li>landscape heterogeneity,</li>
<li>forest cover.</li>
</ol>
<p>If these are available for the complete land mask of pan-EU area, these can then be used to assess soil degradation state (e.g.&nbsp;salinization / sealing level and trends, concentration of excess nutrients and pollutants and trends, erosion state and trends, loss of SOC and trends etc), current soil properties and soil potential in terms of potential ecosystem services / potential SOC sequestration etc, potential productivity of soil, potential soil biodiversity etc. The working version of what we find as feasible to map at high spatial resolution is provided below.</p>
<p>Note that some of the soil health indicators recommended by the European Commission / JRC, are not <em>defacto</em> soil variables (e.g.&nbsp;vegetation cover, landscape heterogeneity, forest cover) but will be generated in this project and integrated into SHDC4EU. Some potential options to represent the vegetation cover, landscape heterogeneity etc include:</p>
<ul>
<li><strong><a href="https://glad.earthengine.app/view/europe-tree-dynamics">Canopy height</a></strong>, data already available for EU but could also be improved by outputs from the Open-Earth-Monitor project,</li>
<li>GPP, <strong>Gross Primary Productivity</strong> (bimonthly and/or annual; based on the last 2–5yrs),</li>
<li>Land cover and land use / <strong>cropping systems and applications</strong> (categories e.g.&nbsp;based on the EuroCrops), producing cropping system maps for the EU;</li>
</ul>
<p>However, producing <a href="https://joint-research-centre.ec.europa.eu/jrc-news-and-updates/eu-crop-map-2021-10-18_en">cropping system maps</a> for the EU, even only for the recent year is not trivial and producing such data should is at the moment optional.</p>
<p>List of target soil variables that will be delivered in the SHDC4EU includes dynamic soil properties covering period 2000–2025+:</p>
<ul>
<li><strong>Soil organic carbon density</strong> (kg/m3) ISO 10694. 1995 for 0–20, 20–50, 50–100 cm depth intervals, 5–year estimates;</li>
<li><strong>Soil carbon stock change</strong> (t/ha/yr) ISO 10694. 1996 for 0–20, 20–50, 50–100 cm depth intervals; long-term</li>
<li><strong>Soil pH in a suspension of soil in water</strong> (-) ISO 10390. 1994 for 0–20, 20–50, 50–100 depth intervals, 5–year estimates;</li>
<li><strong>Soil pH measured in a CaCl2 solution</strong> (-) ISO 10390. 1994 for 0–20, 20–50, 50–100 cm depth intervals, 5–year;</li>
<li><strong>Soil total nitrogen content</strong> (dg/kg) ISO 11261:1995 for 0–20, 20–50, 50–100 cm depth intervals, 5–year;</li>
<li><strong>Soil bulk density</strong> (t/m3) Adapted ISO 11272:2017 for 0–20, 20–50, 50–100 cm, 5–year;</li>
<li><strong>Soil texture fractions</strong> (sand, silt, clay) (g/g) ISO:11277 for 0–20, 20–50, 50–100 cm depth intervals, 5–year;</li>
<li><strong>Soil WRB subgroup</strong> (factor / probs) based on the WRB2022 classificaton system for 0–200 cm, long-term;</li>
<li><strong>Depth to bedrock</strong> (cm) up to 200 cm, long-term;</li>
<li><strong>Extractable potassium content</strong> (mg/kg) USDA−NRCS, 2004 for 0–20, 20–50, 50–100 cm depth intervals, 5–year;</li>
<li><strong>Carbonates content CaCO3</strong> (g/g) ISO 10693:1995 for 0–20, 20–50, 50–100 cm depth intervals, 5–year;</li>
<li><strong>Extractable Phosphorus content (Olsen)</strong> (mg/kg) ISO 11263. 1994 for 0–20, 20–50, 50–100 cm depth intervals, 5–year;</li>
<li><strong>Monthly Gross Primary Productivity</strong> (kg/ha/yr) FluxNET bi-monthly;</li>
<li><strong>Bare Soil Fraction</strong> (m2/m2) Landsat-based, 5–year;</li>
</ul>
</section>
<section id="variable-registry" class="level2">
<h2 class="anchored" data-anchor-id="variable-registry">Variable registry</h2>
<p>For each variable we will organize a soil variable registry system, so that each variable should have unique (short-as-possible) code. As simple coding system for SHDC4EU is suggested where the variable name is a combination of the three components:</p>
<ul>
<li>generic variable name,</li>
<li>specific laboratory / field / O&amp;M method (ISO standard or similar),</li>
<li>measurement unit,</li>
</ul>
<p>For example, for SOC content in weight percent we recommend using <code>oc_iso.10694.1995_wpct</code> (dry combustion) or <code>oc_iso.17184.2014_wpct</code>; for SOC in permiles you can use e.g.&nbsp;<code>oc_iso.10694.1995_wpml</code>. Note that here we use ISO code, however since ISO is highly commercial and not-practical for open datasets, we recommend using the ISO code (allowed), but in fact pointing to a scientific reference i.e.&nbsp;a publication or PDF that is at the order of scale easier to obtain (instead of pointing to ISO website or similar). For example the variable <code>oc_iso.10694.1995_wpct</code> can be linked to the scientific reference <a href="https://acsess.onlinelibrary.wiley.com/doi/abs/10.2134/agronmonogr9.2.2ed.c29">Nelson et al.&nbsp;(1982)</a> (also used by ISO).</p>
<p>Soil variable names can be also provided in a shorter version (assuming that only 1 reference method exists) as a combination of the variable name and measurement unit code e.g.&nbsp;<code>oc.wcpt</code>. It is very important that all partners in the project consistently use the same codes, and if there are updates in the naming convention that they refer to the version of the code.</p>
</section>
<section id="spatio-temporal-reference" class="level2">
<h2 class="anchored" data-anchor-id="spatio-temporal-reference">Spatio-temporal reference</h2>
<p>The general interest of this project is to produce a time-series of predictions of key soil/vegetation/land use variables with either annual or monthly support and covering the period 1997–2022+ and at highest spatial resolution e.g.&nbsp;30-m. For variables that vary by depth, we will predict at 5 standard depths (0, 20, 50, 100 and 200 cm), then aggregate values to standard depth intervals:</p>
<ul>
<li>0–20 cm (<strong>topsoil</strong>) LUCAS standard;</li>
<li>20–50 cm (<strong>subsoil1</strong>);</li>
<li>50–100 cm (<strong>subsoil2</strong>);</li>
<li>100–200 cm (<strong>subsoil3</strong>);</li>
</ul>
<p>Temporal support of predictions / simulations can be one of the following:</p>
<ul>
<li>Long-term primary soil properties and classes: e.g.&nbsp;soil types, soil water holding capacity;</li>
<li>Annual to 5–year values of primary soil and vegetation variables: e.g.&nbsp;annual bare soil fraction (index);</li>
<li>Bimonthly, monthly or seasonal values of soil and vegetation variables: e.g.&nbsp;bimonthly GPP.</li>
<li>Weekly, 16–day values (original Landsat GLAD),</li>
<li>Daily or hourly values (probably not of interest in this project).</li>
</ul>
<p>Aggregated values of target soil and vegetation variables can also refer to some part of the distribution e.g.&nbsp;quantile e.g.&nbsp;P25, P50 (median / mean value) and P75.</p>
</section>
<section id="uncertainty-of-predictions" class="level2">
<h2 class="anchored" data-anchor-id="uncertainty-of-predictions">Uncertainty of predictions</h2>
<p>As an ideal case, each predictive mapping model should produce / provide also uncertainty per pixel (prediction error) and summary results of robust cross-validation. Two options are possible for predictive error mapping:</p>
<ul>
<li><strong>Provide lower and upper quantiles</strong> <code>p=0.05</code> and <code>p=0.95</code> so that a 90% probability prediction interval can be derived. For variables with skewed distribution / log-normally distributed it is recommended that the lower values are computed in the log-transformed space to avoid predicting negative values.</li>
<li><strong>Provide prediction error as 1 standard deviation</strong>: from this also prediction interval can be derived, assuming that the variable is normally distributed.</li>
</ul>
<p>As a general recommendation we suggest using the <strong>Conformal Prediction method</strong> to produce prediction errors per pixel, best as implemented in the python libraries <a href="https://mapie.readthedocs.io/en/latest/">mappie</a> and/or <a href="https://github.com/deel-ai/puncc">puncc</a>.</p>
</section>
<section id="back-end-front-end-components" class="level2">
<h2 class="anchored" data-anchor-id="back-end-front-end-components">Back-end / front-end components</h2>
<p>The back-end of the SHDC4EU is based on using open source software such as <a href="https://postgis.net/">PostGIS/PostGreSQL</a> <a href="https://cogeotiff.github.io/rio-tiler/">Rio-tiler</a>, <a href="https://fastapi.tiangolo.com/">FastAPI</a> and S3 in the back-end; <a href="https://vuejs.org/">Vue.js</a>, OpenLayers and Geoserver in front end. We are aiming at using / building upon simple scalable solutions built on top of existing open source solutions. Note that the most important about this back-end design is that the system will be (A) <strong>cloud-native</strong> i.e.&nbsp;building upon cloud-optimized solutions, (B) easy to extend, (C) focused on usability of data i.e.&nbsp;serving seamless layers (complete, consistent, documented, version-controlled with a live support channels).</p>
<p>All pan-European layers (COGs) produced in this project, including majority of covariate layers, will be distributed through an existing infrastructure <a href="http://EcoDataCube.eu">http://EcoDataCube.eu</a> (maintained by OpenGeoHub foundation). This means that all layers will be made available:</p>
<ul>
<li>For viewing i.e.&nbsp;as <strong>Web Mapping Service</strong> (WMS) so it can be displayed using OpenLayers or similar;</li>
<li>For direct data access via <strong>Simple Storage Service</strong> (S3) / files registered via the <strong>SpatioTemporal Asset Catalog</strong> (STAC) via <a href="https://stac.ecodatacube.eu">https://stac.ecodatacube.eu</a>;</li>
<li>Back-up copy available via the <strong>NextCloud</strong> including via <a href="https://kwb-r.github.io/kwb.nextcloud/">WebDAV</a>.</li>
</ul>
<p>OpenGeoHub will be responsible that the data is available openly in-real-time (i.e.&nbsp;through S3 service) up to 5 years after the end of project, and that a copy of the data is archived on Zenodo or similar.</p>
<p>WP5 will focus exclusively on pan-EU modeling, however, we might also use national / pilot data to test methods and develop experimental solutions that potentially could have much higher impact. For this purpose it is crucial that all project partners have access to the NextCloud and can freely collaborate on data without a need to have multiple copies and installations.</p>
<p>All tabular vector data will be stored in a <strong>single PostGIS DB</strong> and where possible made available to project participants (a GeoPKG copy of the point / polygon data will also be made available via the project Nextcloud). For land mask / geospatial standards we will rely on the <a href="https://land.copernicus.eu/pan-european">Copernicus Land monitoring infrastructure</a> so we will also include Turkey / Western Balkans and Ukraine i.e.&nbsp;also all candidate countries. For data exchange and sharing we use Nextcloud or similar. This data is however not publicly available.</p>
</section>
<section id="data-exchange-formats" class="level2">
<h2 class="anchored" data-anchor-id="data-exchange-formats">Data exchange formats</h2>
<p>All project partners agree to use standard formats to exchange data within the project and on the project websites. The recommended file formats for gridded spatial / spatiotemporal data include:</p>
<ul>
<li><a href="https://www.cogeo.org/">Cloud-Optimized GeoTIFF</a> (<code>*.tif</code>).</li>
<li><a href="https://gdal.org/drivers/raster/zarr.html">Zarr</a> (<code>*.zarr</code>) (also in combination with <a href="https://gdal.org/drivers/raster/netcdf.html">NetCDF</a> (<code>*.nc</code>).</li>
</ul>
<p>For vector geospatial data we recommend the following file formats:</p>
<ul>
<li>For data subsets: <a href="https://geojson.org/">GeoJSON</a> and/or KML.</li>
<li>For cloud-data serving: <a href="https://github.com/opengeospatial/geoparquet">Geoparquet</a> (<code>*.geoparquet</code>) and/or <a href="https://github.com/eto-ai/lance">lance</a>.</li>
<li>For visualization: <a href="https://gdal.org/drivers/vector/mvt.html">Mapbox Vector Tiles</a> (<code>*.mvt</code>).</li>
<li>For GIS analysis: <a href="https://www.geopackage.org/">Geopackage</a> (<code>*.gpkg</code>).</li>
</ul>
<p>For tabular data, objects as <a href="https://en.wikipedia.org/wiki/Simple_Features">Simple Features</a> and similar, the following file formats are accepted for delivery:</p>
<ul>
<li>Comma Separated Value (<code>*.csv</code>) and <a href="https://giswiki.hsr.ch/GeoCSV">GeoCSV</a> best compressed as <code>*.csv.gz</code>.</li>
<li>R RDS (<code>*.rds</code>) and/or <a href="https://github.com/traversc/qs">QS</a> files that can be <a href="http://mgimond.github.io/ES218/Week02b.html">red and written in parallel</a>.</li>
<li><a href="https://gdal.org/drivers/vector/geojson.html#vector-geojson">GeoJSON</a> (<code>*.json</code>).</li>
<li><a href="https://github.com/flatgeobuf/flatgeobuf">FlatGeobuf</a> (<code>*.fgb</code>).</li>
</ul>
<p>For geospatial data to be FAIR, it should at least pass the following checks:</p>
<ul>
<li>It is decision-ready, or at least analysis-ready (complete consistent optimized);</li>
<li>It is available in a professional catalog e.g.&nbsp;<a href="https://stacspec.org/">STAC catalog</a> and/or <a href="https://geonetwork-opensource.org/">Geonetwork</a>;</li>
<li>It comes with technical documentation (ideally a peer-reviewed publication) / links to Github / Gitlab where users can find technical explanation of how was the data produced;</li>
<li>It has a version and each version has unique <a href="https://www.doi.org/">DOI</a>;</li>
<li>It can be accessed directly i.e.&nbsp;file URL is available for HTTP requests (through S3 or similar).</li>
</ul>
<p>For Cloud-Optimized GeoTIFFs (COG’s) it is highly recommended that all files are prepared using recommended settings i.e.&nbsp;a <a href="https://github.com/cogeotiff/cog-spec/blob/master/spec.md">command line</a> (&gt;GDALv3.2):</p>
<pre><code>gdal_translate in.vrt out.tif -co TILED=YES -co COPY_SRC_OVERVIEWS=YES -co COMPRESS=LZW</code></pre>
<p>Each COG will be quality controlled using (1) <a href="https://github.com/rouault/cog_validator">COG validator</a>, (2) by visual inspection, (3) random sampling point overlay to certify that &gt;99% of pixels are available. After the quality control, all produced global mosaics will be registered and uploaded to S3 storage or similar.</p>
</section>
<section id="standard-spatialtemporal-resolutions-and-support-sizes" class="level2">
<h2 class="anchored" data-anchor-id="standard-spatialtemporal-resolutions-and-support-sizes">Standard spatial/temporal resolutions and support sizes</h2>
<p>For the sake of consistency and compatibility, project participants will use standard spatial resolutions to deliver and exchange data. Recommended standard pixel sizes / resolutions:</p>
<p><em>Table: Standard bounding box and spatial resolutions.</em></p>
<table>
<tbody><tr>
<td>
Europe COG<br>
Bounding box
</td>
<td>
Continental EU COG based on <a href="https://land.copernicus.eu/portal_vocabularies/geotags/eea39">Copernicus</a><br>
Spatial resolutions and image size <a href="https://epsg.io/3035">EPSG:3035</a>
</td>
</tr>
<tr>
<td>
Xmin = 900,000
<p>
Ymin = 899,000
</p><p>
Xmax = 7,401,000
</p><p>
Ymax = 5,501,000
</p></td>
<td>
10m | 650,100L x 460,200P
<p>
25m | 260,040L x 184,080P
</p><p>
30m | 216,700P x 153,400L
</p><p>
100m | 65,010P x 46,020L
</p><p>
250m | 26,004P x 18,408L
</p><p>
1km | 6501P x 4602L
</p></td>
</tr>
</tbody></table>
</section>
<section id="file-naming-convention" class="level2">
<h2 class="anchored" data-anchor-id="file-naming-convention">File Naming convention</h2>
<p>SHDC will consistently use the standard OpenLandMap file-naming convention to submit new data-sets and similar (compare e.g.&nbsp;with the MODIS file naming convention). This is to ensure consistency and ease of use within the AI4SoilHealth project, but also by the end-users. This applies especially to WP4, WP5 and WP6.</p>
<p>The OpenLandMap file-naming convention works with 10 fields that basically define the most important properties of the data (this way users can search files, prepare data analysis etc, without even needing to access or open files. The 10 fields include:</p>
<ol type="1">
<li>Generic variable name (needs to be unique and approved by the AI4SoilHealth Coordination team): <code>lclu</code>;</li>
<li>Variable procedure combination i.e.&nbsp;method standard (standard abbreviation): <code>luisa</code>;</li>
<li>Position in the probability distribution / variable type: <code>c</code>;</li>
<li>Spatial support (usually horizontal block) in m or km: <code>30m</code>;</li>
<li>Depth reference or depth interval e.g.&nbsp;below (“b”), above (“a”) ground or at surface (“s”): <code>s</code>;</li>
<li>Time reference begin time (YYYYMMDD): <code>20210101</code>;</li>
<li>Time reference end time: <code>20211231</code>;</li>
<li>Bounding box (2 letters max): <code>eu</code>;</li>
<li>EPSG code: <code>epsg.3035</code>;</li>
<li>Version code i.e.&nbsp;creation date: <code>v20221015</code>;</li>
</ol>
<p>An example of a file-name based on the description above:</p>
<pre><code>lclu_luisa_c_30m_s_20210101_20211231_eu_epsg.3035_v20221015.tif</code></pre>
<p>Note that this file naming convention has the following properties:</p>
<ul>
<li>Large quantities of files can be easily sorted and searched (one line queries in <a href="https://www.gnu.org/software/bash/">Bash</a>).</li>
<li>File-naming patterns can be used to seamlessly build virtual mosaics and composites.<br>
</li>
<li>Key spatiotemporal properties of the data are available in the file name e.g.&nbsp;variable type, O&amp;M method, spatial resolution, bounding box, projection system, temporal references. Users can program analysis without opening or testing files.</li>
<li>Versioning system is ubiquitous.</li>
<li>All file-names are unique.</li>
</ul>
<p>Geonetwork and STAC will be further used to link the unique file names to: (1) WPs, deliverables, themes / keywords, (2) DOI’s, (3) project homepages, (4) contact pages for support and feedback. For keywords we recommend using the <a href="https://inspire.ec.europa.eu/glossary">INSPIRE keywords</a>. To confirm that metadata is complete and consistent, we recommend using the <a href="https://inspire.ec.europa.eu/validator/home/index.html">INSPIRE metadata validator</a> and/or <a href="https://data.europa.eu/en">https://data.europa.eu/en</a> validator.</p>
<p>Some simple additional rules for generating the file name include:</p>
<ol type="1">
<li>Codes and abbreviations should be human-readable as much as possible (hence short, but not too short!);</li>
<li>Use only English-US (en-us) language e.g.&nbsp;for months use <code>jan</code>, <code>feb</code> etc;</li>
<li>Consistently use <a href="https://unicode.org/standard/standard.html">UNICODE standard</a>: small letters only, no blank spaces, no non-ASCII characters;</li>
<li>Limit the total file name size in characters to <code>256</code>;</li>
<li>For time reference do not extend beyond hour minute and timezone;</li>
<li>For bounding boxes use as much as possible the 2–letter unique country code; for continents use the Equi7 Grid code i.e.&nbsp;<code>eu</code>,</li>
<li>For method codes use as much as possible unique IDs from <a href="https://www.iso.org/standards-catalogue/browse-by-ics.html">ISO - ICS</a>;</li>
<li>For MODIS products use consistently the MODIS products codes e.g.&nbsp;<a href="https://lpdaac.usgs.gov/products/mod11a2v061/">MOD11A2 v061</a>; For long-term aggregates of seasonal, monthly, weekly values use the period name at the end of the method names (#3) for example the long-term estimate of MODIS LST daytime temperature for month August:</li>
</ol>
<pre><code>lst.d_mod11a2v061.aug_m_1km_s_20200101_20211231_eu_epsg.3035_v20221015.tif</code></pre>
<p>A list of vocabularies to be used as abbreviated names of variables will be provided by OpenGeoHub. The same file-name convention described above can be also used for vector data (this would only have a different file extension) also.</p>
</section>
<section id="registering-project-outputs" class="level2">
<h2 class="anchored" data-anchor-id="registering-project-outputs">Registering project outputs</h2>
<p>All project participants are required to register project outputs and inform other project participants about the progress (via <a href="https://zenodo.org/communities/ai4soilhealth/">https://zenodo.org/communities/ai4soilhealth/</a>). For internal documents / draft versions of outputs the following submission principles need to be followed:</p>
<ul>
<li>For internal datasets and documents use project Gitlab and Mattermost to inform parties about the progress and receive feedback;</li>
<li>Regularly inform your WP about the progress and planned dates (best via Mattermost channel).</li>
<li>To receive internal feedback please use project Gitlab; this way ALL project participants can follow discussion and can possibly help resolve bugs / open issues.</li>
<li>For each new version of the output specify in short terms what has been improved, what has changed from last version.</li>
<li>Tag people in Mattermost that you would like to request to review the outputs.</li>
</ul>
<p>For official releases please refer to the Consortium Agreement for the correct procedure. For every public release of the new official AI4SoilHealth data set, functionality and/or software, the following procedure should be followed closely:</p>
<ul>
<li>First register the produced data set, library, front-end via project management system (Gitlab) and confirm that the submission follows the <strong>minimum requirements</strong> regarding file formats and file naming, documentation and metadata.</li>
<li>Make sure that no parts of the submission have any <strong>copyright issues</strong>.</li>
<li>Make sure that <strong>all contributors are credited in detail</strong>.</li>
<li>Make sure you provide appropriate <strong>disclaimer</strong> and <strong>terms of use</strong>; we recommend using “No warranty” as default.</li>
<li>Make sure the <strong>license</strong> is compatible to the AI4SoilHealth project following our Consortium Agreement;</li>
<li>Make sure <strong>acknowledgement</strong> (see below) is clearly provided.</li>
<li>Make sure you provide <strong>contact information</strong> and instructions for users to provide feedback and request support.</li>
<li>Make sure all project partners are informed about the date of release (through Gitlab and Mattermost).</li>
</ul>
<p>Suggested 3rd party platforms for registering outputs (we recommend using multiple of the listed options e.g.&nbsp;a, b, c etc):</p>
<ol type="1">
<li>Datasets:
<ol type="1">
<li><a href="https://zenodo.org/communities/ai4soilhealth/">https://zenodo.org/communities/ai4soilhealth/</a> and/or <a href="https://onedata.org/">https://onedata.org/</a>,</li>
<li>Metadata entry via: Geonetwork and/or STAC,</li>
<li>Structured summary via: <a href="https://ai4soilhealth.eu">https://ai4soilhealth.eu</a>,</li>
<li>Suggestion: registered on <a href="https://data.europa.eu/en">https://data.europa.eu/en</a>,</li>
</ol></li>
<li>Software:
<ol start="5" type="1">
<li>Github (<a href="https://github.com/ai4soilhealth">https://github.com/ai4soilhealth</a>) -&gt; suggestion: always generate DOI and put on Zenodo.org,</li>
<li><a href="https://archive.softwareheritage.org/">https://archive.softwareheritage.org/</a>,</li>
<li>Structured summary via: <a href="https://ai4soilhealth.eu">https://ai4soilhealth.eu</a>,</li>
</ol></li>
<li>Tutorials / data catalogs:
<ol start="8" type="1">
<li>Webpages on github / gitlab (Rbookdown, python books) see e.g.&nbsp;<a href="https://opengeohub.github.io/SoilSamples/">https://opengeohub.github.io/SoilSamples/</a>,</li>
<li>MKdocs see e.g.&nbsp;<a href="https://gee-community-catalog.org/">https://gee-community-catalog.org/</a>;</li>
</ol></li>
<li>Data portals / web-GUI’s
<ol start="10" type="1">
<li>Sub-domain under *.ai4soilhealth.eu,</li>
<li>Recommended portal: <a href="https://gkhub.earthobservations.org/">https://gkhub.earthobservations.org/</a>,</li>
<li>Video-tutorial published via: <a href="https://av.tib.eu/">https://av.tib.eu/</a>,</li>
</ol></li>
</ol>
<p>We are planning to have our own installation of Gitlab, <a href="https://github.com/radiantearth/stac-browser">STAC browser</a>, <a href="https://geonetwork-opensource.org/">Geonetwork</a> and <a href="https://pretalx.com/">Pretalx</a>. Final workflow for registering outputs will be specified in the final version of the implementation plan.</p>
<p>During publishing of outputs, and especially if you register outputs via 3rd party repos, it of utmost importance that all partners use the correct attribution and links to project homepage:</p>
<ul>
<li>Correct attribution: “The AI4SoilHealth project project has received funding from the European Union’s Horizon Europe research an innovation programme under grant agreement <a href="https://cordis.europa.eu/project/id/101086179">No.&nbsp;101086179</a>.”;</li>
<li>Project URL: <a href="https://cordis.europa.eu/project/id/101086179">https://cordis.europa.eu/project/id/101086179</a>.</li>
<li>Project homepage: <a href="https://ai4soilhealth.eu">https://ai4soilhealth.eu</a>,</li>
<li>Correct default disclaimer is below.</li>
</ul>
</section>
<section id="disclaimer-for-data-software-products" class="level2">
<h2 class="anchored" data-anchor-id="disclaimer-for-data-software-products">Disclaimer for data / software products</h2>
<p>All public releases of new data / software should point to the generic AI4SoilHealth disclaimer:</p>
<blockquote class="blockquote">
<p>“Funded by the European Union. Views and opinions expressed are however those of the author(s) only and do not necessarily reflect those of the European Union or European Commision. Neither the European Union nor the granting authority can be held responsible for them. The data is provided “as is”. AI4SoilHealth project consortium and its suppliers and licensors hereby disclaim all warranties of any kind, express or implied, including, without limitation, the warranties of merchantability, fitness for a particular purpose and non-infringement. Neither AI4SoilHealth Consortium nor its suppliers and licensors, makes any warranty that the Website will be error free or that access thereto will be continuous or uninterrupted. You understand that you download from, or otherwise obtain content or services through, the Website at your own discretion and risk.”</p>
</blockquote>
</section>
<section id="land-mask" class="level2">
<h2 class="anchored" data-anchor-id="land-mask">Land mask</h2>
<p>Considering the land mask for pan-EU (see figure), we will closely match the data coverage of <a href="https://land.copernicus.eu/pan-european">Copernicus pan-european</a> i.e.&nbsp;the official selection of countries listed <a href="https://lanEEA39d.copernicus.eu/portal_vocabularies/geotags/eea39">here</a>. Note the mask covers also all EU integration candidate countries, but can eventually also be subset to only European Union countries.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/land_mask_paneu.png" class="img-fluid figure-img" style="width:70.0%"></p>
<figcaption>Landmask for the SHDC for pan-EU.</figcaption>
</figure>
</div>
</div>
</div>
<p>There are a total of three landmask files available, each of which is aligned with the standard spatial/temporal resolution and sizes of SHDC4EU specifications. Additionally, these files include a corresponding look-up table that provides explanations for the values present in the raster data.</p>
<p><em>Table: Technical description of the pan-EU land mask.</em></p>
<table>
<tbody><tr>
<td rowspan="5">
Landmask
</td>
<td>
Purpose/principle
</td>
<td>
The basic principle to create the land mask is to include as much as land as possible, to avoid missing any land pixels and ensure precise differentiation between land, ocean and inland water bodies.
<p>
When generating the land mask, the two reference datasets in a way that:
</p><ul>
<li>
<p>If either of the two reference datasets identifies a pixel as land, it is considered a land pixel in our mask.</p>
</li><li>
Regarding ocean and inland water bodies, a pixel is classified as a water pixel only when both reference datasets confirm its identification as water.
</li>
</ul>
</td>
</tr>
<tr>
<td>
Reference datasets
</td>
<td>
<ol>
<li>
<p><a href="https://esa-worldcover.org/en">WorldCover</a>, 10 m resolution.</p>
</li><li>
<a href="https://www.mapsforeurope.org/datasets/euro-global-map">EuroGlobalMap</a>, with shapefiles of administrative boundaries, inland water bodies, ocean and landmask.
</li>
</ol>
</td>
</tr>
<tr>
<td>
Mosaic/resampling method
</td>
<td>
The coarse resolution landmasks (&gt;10 m) are generated by resampling from the 10m resolution base map using resampling method “min” in GDAL. This “min” method allows taking the minimum values from the contributing pixels, to keep as much land as possible.
</td>
</tr>
<tr>
<td>
Resolution available
</td>
<td>
10-m, 30-m, 100-m, 250-m, and 1-km resolution
</td>
</tr>
<tr>
<td>
Mask values
</td>
<td>
<ul>
<li>
<p>10: not in the pan-EU area, i.e.&nbsp;out of mapping scope</p>
</li><li>
<p>1: land</p>
</li><li>
<p>2: inland water</p>
</li><li>
3: ocean
</li>
</ul>
</td>
</tr>
<tr>
<td rowspan="5">
ISO-3166 country code mask
</td>
<td>
Purpose/principle
</td>
<td>
In this mask, each country is assigned a unique value, which allows for the interpretation and analysis of data associated with a specific country. The values are assigned to each country according to iso-3166 country code, which can be found in the corresponding look-up table.
</td>
</tr>
<tr>
<td>
Reference datasets
</td>
<td>
<a href="https://www.mapsforeurope.org/datasets/euro-global-map">EuroGlobalMap</a> country shapefile
</td>
</tr>
<tr>
<td>
Mosaic/resampling method
</td>
<td>
The coarse resolution masks (&gt;10 m) are generated by resampling from the 10m resolution base map using resampling method “mode” in GDAL.
</td>
</tr>
<tr>
<td>
Resolution available
</td>
<td>
10m, 30m and 100m
</td>
</tr>
<tr>
<td>
Mask values
</td>
<td>
Can be found in the corresponding look-up table.
</td>
</tr>
<tr>
<td rowspan="5">
NUTS-3 mask
</td>
<td>
Purpose/principle
</td>
<td>
In this raster file, each unique NUT3 level area is assigned a unique value, which allows for the interpretation and analysis of data associated with specific NUTS-3 regions. Compared to ISO-3166 country code mask, NUTS-3 mask shows more details about regional administrative boundaries.
</td>
</tr>
<tr>
<td>
Reference datasets
</td>
<td>
European NUTS-3 shapefile
</td>
</tr>
<tr>
<td>
Mosaic/resampling method
</td>
<td>
The coarse resolution masks (&gt;10 m) are generated by resampling from the 10m resolution base map using resampling method “mode” in GDAL.
</td>
</tr>
<tr>
<td>
Resolution available
</td>
<td>
10 m, 30 m and 100 m
</td>
</tr>
<tr>
<td>
Mask values
</td>
<td>
Can be found in the corresponding look-up table.
</td>
</tr>
</tbody></table>
<p>An important point to consider is that the ISO-code country mask provides a wider geographical coverage compared to the NUTS3 mask. This extended coverage includes countries such as Ukraine and others that lie beyond the NUTS3 administrative boundaries. Both the land mask and administrative code mask are in an Equal-area projection, allowing for accurate area estimation and facilitating aggregation per political/administrative unit.</p>
<p>These masks will be published and shared in a publicly available way on Zenodo. The working version (v0.1) is available from: <a href="https://doi.org/10.5281/zenodo.8171860">https://doi.org/10.5281/zenodo.8171860</a>. The layers can be opened directly in QGIS by copying links from Zenodo. The scripts used to generate these masks can be found in our <a href="https://gitlab.opengeohub.org/ai4soilhealth/internal-implementation/-/issues">project Gitlab</a> or via the <a href="https://github.com/AI4SoilHealth/SoilHealthDataCube/">public Github</a>. If any issue / problem is noticed, please report.</p>
</section>
<section id="layer-submission-process-and-quality-control" class="level2">
<h2 class="anchored" data-anchor-id="layer-submission-process-and-quality-control">Layer submission process and quality control</h2>
<p>Partners on consortium are invited to submit working versions of data and services / share preliminary outputs internally via NextCloud or similar. Folder structure and more detailed instructions will be provided by the WP lead OpenGeoHub. It is important however to distinguish between (1) internal releases, (2) public releases (partners responsible) and (3) public releases approved by the consortium.</p>
<p>All WP5 participants are required to register project outputs and inform other project participants about the progress. For internal documents / draft versions of outputs the following submission principles need to be followed:</p>
<ul>
<li>For internal datasets and documents use project repositories, Gitlab, NextCloud and Mattermost to inform parties about the progress and receive feedback;</li>
<li>Regularly inform your WP/Task group about the progress and planned dates (best via Mattermost channel).</li>
<li>To receive internal feedback please use project Gitlab; this way ALL project participants can follow discussion and can possibly help resolve bugs / open issues.</li>
<li>For each new version of the output specify in short terms what has been improved, what has changed from last version.</li>
<li>Tag people in Mattermost that you would like to request to review the outputs.</li>
</ul>
<p>For official releases please refer to these guidelines to avoid any delays. For every public release of the new official AI4SoilHealth data set, functionality and/or software, the following procedure should be followed closely:</p>
<ul>
<li>First register the produced data set, library, front-end via project management system (Gitlab) and confirm that the submission follows the <strong>minimum requirements</strong> regarding file formats and file naming, documentation and metadata.</li>
<li>Make sure that no parts of the submission have any <strong>copyright issues</strong>.</li>
<li>Make sure that <strong>all contributors are credited in detail</strong>.</li>
<li>Make sure you provide appropriate <strong>disclaimer</strong> and <strong>terms of use</strong>; we recommend using “No warranty” as default.</li>
<li>Make sure the <strong>license</strong> is compatible to the AI4SoilHealth project following our Consortium Agreement;</li>
<li>Make sure <strong>acknowledgement</strong> (<a href="https://cordis.europa.eu/project/id/101086179">https://cordis.europa.eu/project/id/101086179</a>) is clearly provided.</li>
<li>Make sure you provide <strong>contact information</strong> and instructions for users to provide feedback and request support.</li>
<li>Make sure all project partners are informed about the date of release (through Gitlab and Mattermost).</li>
<li>If there are any deviations from specifications you have put in the implementation plan, please indicate such deviations in the project management system.</li>
</ul>
<p>For quality assurance (public releases) we recommend the following three checks for any new significant datasets:</p>
<ol type="1">
<li>Preliminary data and code shared with all WP members via NextCloud / code on Gitlab / Github. Internal check in co-development sessions.</li>
<li>Peer-review publications (as agreed in the proposal, all major new data products should go through peer-review).</li>
<li>Data and code exposed for public comments (<a href="https://github.com/ai4soilhealth">https://github.com/ai4soilhealth</a>) including the social media, especially Mastodon, X, Linkedin etc.</li>
</ol>
<p>In principle, the publication process should be non-bureaucratic, agile and should not limit intonation and experimentation. However, for public releases approved by the consortium e.g.&nbsp;major deliverables as specified in the Grant Agreement, it is advised that these are approved by the Executive Board of the project so that the new dataset / service is then also officially promoted through media channels, landing page etc.</p>
</section>
<section id="reproducible-research" class="level2">
<h2 class="anchored" data-anchor-id="reproducible-research">Reproducible research</h2>
<p>In order to ensure FAIR outputs, it is highly recommended that all production steps used to generate maps are documented in code, best as <strong>Rmarkdown / Python Jupyter computational notebooks</strong>. As agreed also in the project proposal, All empirical data obtained will follow strict and validated processes of monitoring and evaluation to avoid any potential error. The data produced in AI4SoilHealth will be made available as open data, following the <strong>FAIR </strong>principle. The project <strong>Data Management Plan (DMP)</strong> is available and has even more instructions.</p>
<p>AI4SoilHealth has agreed to provide <strong>open access (OA)</strong> to research outputs (<em>e.g.</em>, publications, data, software) through deposition in trusted repositories. Partners will provide OA for peer-reviewed scientific publications relating to their results. Authors of all peer-reviewed scientific publications will store them in an OA trusted repository, during and after the project’s life following Article 17 and Annex 5 of the General Assembly. The consortium members will be encouraged to publish in the Open Research Europe data platform, specifically via the Zenodo community for AI4SoilHealth (<a href="https://zenodo.org/communities/ai4soilhealth/">https://zenodo.org/communities/ai4soilhealth/</a>).</p>
<p>Besides OA publication, the project aims for <strong>early and open sharing of the soil health data, and the research and technological developments </strong>including open data, open standards, open source software, and open communication:</p>
<ul>
<li><strong>Open data</strong>: The AI4SoilHealth consortium will build solutions upon open datasets published using genuinely <a href="https://opendefinition.org/licenses/">open data licenses</a>.<br>
</li>
<li><strong>Open standards (interoperability)</strong>: The use of open standards prevents lock-in by, or dependency on any single data, software or service supplier. AI4SoilHealth will fully adopt the principles of FAIR data management and open standards to enable interoperability of methods and services, both within the project and beyond.</li>
<li><strong>Open source software</strong>: AI4SoilHealth plans to release the mobile phone application on Github under the MIT license. All major outputs of the project should be made available via the project github (<a href="https://github.com/ai4soilhealth">https://github.com/ai4soilhealth</a>).</li>
<li><strong>Open communication:</strong> The consortium has experience with organizing live, open, and free discussion forums and workshops.</li>
</ul>



</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-estel2016mapping" class="csl-entry" role="listitem">
Estel, Stephan, Tobias Kuemmerle, Christian Levers, Matthias Baumann, and Patrick Hostert. 2016. <span>“<span class="nocase">Mapping cropland-use intensity across Europe using MODIS NDVI time series</span>.”</span> <em>Environmental Research Letters</em> 11 (2): 024015. <a href="https://doi.org/10.1088/1748-9326/11/2/024015">https://doi.org/10.1088/1748-9326/11/2/024015</a>.
</div>
<div id="ref-ettehadi2019separating" class="csl-entry" role="listitem">
Ettehadi Osgouei, Paria, Sinasi Kaya, Elif Sertel, and Ugur Alganci. 2019. <span>“Separating Built-up Areas from Bare Land in Mediterranean Cities Using Sentinel-2A Imagery.”</span> <em>Remote Sensing</em> 11 (3): 345. <a href="https://doi.org/10.3390/rs11030345">https://doi.org/10.3390/rs11030345</a>.
</div>
<div id="ref-hengl2024osg" class="csl-entry" role="listitem">
Hengl, T. 2024. <span>“Soil Health Monitoring Through Iterative Analysis of Soil’s Past, Present, and Future.”</span> <em>Open Access Government</em> 8 (April 2024): 1–2. <a href="https://doi.org/10.56367/OAG-042-11173">https://doi.org/10.56367/OAG-042-11173</a>.
</div>
<div id="ref-mzid2021analysis" class="csl-entry" role="listitem">
Mzid, Nada, Stefano Pignatti, Wenjiang Huang, and Raffaele Casa. 2021. <span>“An Analysis of Bare Soil Occurrence in Arable Croplands for Remote Sensing Topsoil Applications.”</span> <em>Remote Sensing</em> 13 (3): 474. <a href="https://doi.org/10.3390/rs13030474">https://doi.org/10.3390/rs13030474</a>.
</div>
<div id="ref-orgiazzi2018lucas" class="csl-entry" role="listitem">
Orgiazzi, Alberto, Cristiano Ballabio, Panagiotis Panagos, Arwyn Jones, and Oihane Fernández-Ugalde. 2018. <span>“<span class="nocase">LUCAS Soil, the largest expandable soil dataset for Europe: a review</span>.”</span> <em>European Journal of Soil Science</em> 69 (1): 140–53. <a href="https://doi.org/10.1111/ejss.12499">https://doi.org/10.1111/ejss.12499</a>.
</div>
<div id="ref-witjes2023ecodatacube" class="csl-entry" role="listitem">
Witjes, Martijn, Leandro Parente, Josip Križan, Tomislav Hengl, and Luka Antonić. 2023. <span>“Ecodatacube. Eu: Analysis-Ready Open Environmental Data Cube for Europe.”</span> <em>PeerJ</em> 11: e15478. <a href="https://doi.org/10.7717/peerj.15478">https://doi.org/10.7717/peerj.15478</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2024, OpenGeoHub foundation</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/AI4SoilHealth/SHDC4EU_manual/edit/main/1-specifications.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/AI4SoilHealth/SHDC4EU_manual/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/AI4SoilHealth/SHDC4EU_manual/">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://fosstodon.org/@opengeohub">
      <i class="bi bi-mastodon" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/AI4SoilHealth">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://www.linkedin.com/company/opengeohub-foundation/">
      <i class="bi bi-linkedin" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://www.youtube.com/c/OpenGeoHubFoundation">
      <i class="bi bi-youtube" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>